modules/Evse15118D20/charger/session_logger.cpp

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+#include "session_logger.hpp"
+
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+#include <stdexcept>
+
+#include <date/date.h>
+
+#include <iso15118/session/logger.hpp>
+
+#include <everest/logging.hpp>
+
+using LogEvent = iso15118::session::logging::Event;
+
+std::string get_filename_for_current_time() {
+    const auto now = std::chrono::system_clock::now();
+    const auto now_t = std::chrono::system_clock::to_time_t(now);
+
+    std::tm now_tm;
+    gmtime_r(&now_t, &now_tm);
+
+    char buffer[64];
+    strftime(buffer, sizeof(buffer), "%y%m%d_%H-%M-%S.yaml", &now_tm);
+    return buffer;
+}
```

Comment:
this could go into a utility lib in lib/staging (but that can be done in another PR where we collect some of the utility functions that are defined in multiple modules)

Reply:
I'll make a note!